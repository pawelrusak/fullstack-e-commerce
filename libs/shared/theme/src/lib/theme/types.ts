export type * from './tokens/types';
export type * from './theme.types';
