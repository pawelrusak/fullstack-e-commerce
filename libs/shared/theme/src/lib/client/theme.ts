import { color, fontFamily, fontWeight, space, screens } from './system-theme';
import { component } from './component-theme';

import type { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  color,
  fontFamily,
  fontWeight,
  space,
  screens,
  component,
};

export default theme;
