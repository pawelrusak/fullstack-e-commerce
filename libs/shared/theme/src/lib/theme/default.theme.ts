import {
  color,
  fontFamily,
  fontWeight,
  space,
  screens,
  component,
} from './tokens';

import type { DefaultTheme } from 'styled-components';

export const defaultTheme: DefaultTheme = {
  color,
  fontFamily,
  fontWeight,
  space,
  screens,
  component,
};
