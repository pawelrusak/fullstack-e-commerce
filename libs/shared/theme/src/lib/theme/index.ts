'use client';

export { default as theme } from './default.theme';
