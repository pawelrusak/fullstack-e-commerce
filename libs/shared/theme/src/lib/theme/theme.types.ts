import type {
  AppSpace,
  AppColor,
  AppFontFamily,
  AppFontWeight,
  AppScreens,
  AppComponent,
} from './tokens';

export type AppSpacesTheme = {
  space: AppSpace;
};

export type AppColorTheme = {
  color: AppColor;
};

export type AppFontFamilyTheme = {
  fontFamily: AppFontFamily;
};

export type AppFontWeightTheme = {
  fontWeight: AppFontWeight;
};

export type AppScreensTheme = {
  screens: AppScreens;
};

export type AppComponentTheme = {
  component: AppComponent;
};

export type AppTheme = AppColorTheme &
  AppFontFamilyTheme &
  AppFontWeightTheme &
  AppSpacesTheme &
  AppScreensTheme &
  AppComponentTheme;
