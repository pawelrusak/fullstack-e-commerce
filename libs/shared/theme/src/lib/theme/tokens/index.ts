export { component } from './component.tokens';
export { color, fontFamily, fontWeight, screens, space } from './system.tokens';
