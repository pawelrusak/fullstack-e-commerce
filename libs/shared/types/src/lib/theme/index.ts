export type * from './theme';
export type {
  BrandComponent,
  ButtonComponent,
  ButtonComponentPalette,
} from './component';
