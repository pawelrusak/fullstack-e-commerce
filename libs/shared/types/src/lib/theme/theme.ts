import { Theme as StyledSystemTheme } from 'styled-system';

// TODO: Move this file to the theme library
export type AppSpacesTheme = {
  space: StyledSystemTheme['space'];
};

export type AppColorTheme = {
  color: {
    primary: string;
    secondary: string;
    text: string;
    background: string;
    // TODO: Rename this to "secondarySurface". "Surface" will be used for UI components' backgrounds, such as cards, modals, navbar, etc.
    secondBackground: string;
    border: string;
    error: string;
  };
};

export type AppFontFamily = {
  fontFamily: {
    /**
     * @deprecated Use "primary" instead
     */
    montserrat: string;
    /**
     * @deprecated Use "secondary" instead.
     */
    poppins: string;
    /**
     * For body text, forms, etc.
     *
     * @warning Use only for headings at level H5 or lower.
     */
    primary?: string;
    /**
     * For headings, titles, etc.
     */
    secondary?: string;
  };
};

export type AppFontWeight = {
  fontWeight: {
    regular: number;
    semiBold: number;
    bold: number;
  };
};

export type AppScreensTheme = {
  screens: {
    sm: string;
    md: string;
    lg: string;
    xl: string;
    xxl: string;
  };
};

export type AppTheme = AppColorTheme &
  AppFontFamily &
  AppFontWeight &
  AppSpacesTheme &
  AppScreensTheme;
