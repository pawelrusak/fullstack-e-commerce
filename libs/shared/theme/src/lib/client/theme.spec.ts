import theme from './theme';

describe('Theme config', () => {
  it('should have the correct structure and types', () => {
    expect(theme).toEqual({
      color: expect.any(Object),
      fontFamily: expect.any(Object),
      fontWeight: expect.any(Object),
      space: expect.arrayContaining([expect.any(Number)]),
      screens: expect.any(Object),
      component: expect.any(Object),
    });
  });

  it('should match the specified values', () => {
    expect(theme).toMatchSnapshot();
  });
});
