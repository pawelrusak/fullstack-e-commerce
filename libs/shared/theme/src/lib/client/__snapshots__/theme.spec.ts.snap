// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme config should match the specified values 1`] = `
{
  "color": {
    "background": "hsl(0, 0%, 100%)",
    "border": "hsl(0, 0%, 96%)",
    "error": "hsl(348, 83%, 47%)",
    "onBackground": "hsl(0, 0%, 19%)",
    "onBackgroundInteract": "hsl(7, 100%, 65%)",
    "onBackgroundLink": "hsl(7, 100%, 65%)",
    "onBackgroundLinkInteract": "hsl(7, 86%, 51%)",
    "onPrimary": "hsl(0, 0%, 100%)",
    "onSecondary": "hsl(0, 0%, 100%)",
    "onSecondarySurface": "hsl(0, 0%, 19%)",
    "onSecondarySurfaceVariant": "hsl(7, 100%, 65%)",
    "onSurface": "hsl(0, 0%, 19%);",
    "onSurfaceInteract": "hsl(7, 100%, 65%)",
    "onSurfaceVariant": "hsl(7, 100%, 65%)",
    "primary": "hsl(7, 100%, 65%)",
    "primaryBorder": "hsl(7, 100%, 65%)",
    "primaryBorderInteract": "hsl(7, 86%, 51%)",
    "primaryInteract": "hsl(7, 86%, 51%)",
    "secondBackground": "hsl(0, 0%, 96%)",
    "secondary": "hsl(0, 0%, 19%)",
    "secondaryBorder": "hsl(0, 0%, 19%)",
    "secondaryBorderInteract": "hsl(0, 0%, 16%)",
    "secondaryInteract": "hsl(0, 0%, 16%)",
    "secondarySurface": "hsl(0, 0%, 96%)",
    "surface": "hsl(0, 0%, 100%)",
    "tertiary": "hsl(46, 99%, 65%)",
    "text": "hsl(0, 0%, 19%)",
  },
  "component": {
    "brand": {
      "base": {
        "svgDot": {
          "fill": "hsl(7, 100%, 65%)",
        },
        "svgText": {
          "fill": "hsl(0, 0%, 19%)",
        },
      },
    },
  },
  "fontFamily": {
    "montserrat": "Montserrat, sans-serif",
    "poppins": "Poppins, sans-serif",
    "primary": "Montserrat, sans-serif",
    "secondary": "Poppins, sans-serif",
  },
  "fontWeight": {
    "bold": 700,
    "regular": 400,
    "semiBold": 600,
  },
  "screens": {
    "lg": "1024px",
    "md": "768px",
    "sm": "640px",
    "xl": "1280px",
    "xxl": "1536px",
  },
  "space": [
    0,
    0.2,
    0.4,
    0.8,
    1.6,
    2.4,
    3.2,
  ],
}
`;
